backend/app/api/src/helpers/AuthenticatedStructures.ts

```diff
@@ -125,44 +127,135 @@ export class AuthenticatedStructures {
+                    if(canAccessPrivateData) {
+                        organizationIdsToGetWebshopsFor.push(organizationId);
+                    }
+                    organizationData.set(organizationId, {organizationRegistrationPeriod, canAccessPrivateData});
+                }
+            }
         }
+        //#endregion
         
-        return OrganizationStruct.create({
-            ...organization.getBaseStructure(),
-            period: await this.organizationRegistrationPeriod(organizationPeriod)
-        })
-    }
+        //#region get periods
+        const allPeriodIds = periodIdOrganizationsMap.keys();
+        const allPeriods = await RegistrationPeriod.getByIDs(...allPeriodIds);
+        const periodMap = new Map<string, RegistrationPeriod>(allPeriods.map(p => [p.id, p]));
+        //#endregion
+
+        //#region get webshop previews
+        const webshops = organizationIdsToGetWebshopsFor.length > 0 ? await Webshop.where(
+            {
+                organizationId: {
+                    sign: 'IN',
+                    value: organizationIdsToGetWebshopsFor
+                }
+            },
+            { select: Webshop.selectColumnsWithout(undefined, "products", "categories")}
+        ) : [];
 
-    static async organizations(organizations: Organization[]): Promise<OrganizationStruct[]> {
-        // for now simple loop
-        if (organizations.length > 10) {
-            console.warn('Trying to load too many organizations at once: ' + organizations.length)
+        const webshopPreviews = new Map<string, WebshopPreview[]>();
+
+        for (const w of webshops) {
+            if (!await Context.auth.canAccessWebshop(w)) {
+                continue
+            }
+
+            const organizationId = w.organizationId;
+            const array = webshopPreviews.get(organizationId);
+            const preview = WebshopPreview.create(w);
+
+            if(array) {
+                array.push(preview);
+            } else {
+                webshopPreviews.set(organizationId, [preview]);
+            }
         }
+        //#endregion
 
-        const structs: OrganizationStruct[] = [];
-        for (const organization of organizations) {
-            structs.push(await this.organization(organization))
+        //#region create organization structs
+        const results: OrganizationStruct[] = [];
+
+        for(const organization of organizations) {
+            const registrationPeriod = periodMap.get(organization.periodId);
+            if(!registrationPeriod) {
+                console.error('Registration period is undefined.')
+                continue;
+            }
+
+            const organizationId = organization.id;
+            const data = organizationData.get(organizationId);
+            if(data === undefined) {
+                console.error('Organization data is undefined.')
```

Comment:
Het speciale geval dat een vereniging geen organizationRegistrationPeriod heeft wordt niet goed opgevangen. Vroeger werd de methode 'getPeriod' gebruikt, die dit correct opving en de periode aanmaakte als die nog niet bestond (veilig in een queue).